clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java

```diff
@@ -130,14 +133,26 @@ public String getVariables() {
 
   @Override
   public Map<String, Object> getVariablesAsMap() {
-    return jsonMapper.fromJsonAsMap(variables);
+    if (variablesAsMap == null) {
+      variablesAsMap = jsonMapper.fromJsonAsMap(variables);
+    }
+    return variablesAsMap;
   }
 
   @Override
   public <T> T getVariablesAsType(final Class<T> variableType) {
     return jsonMapper.fromJson(variables, variableType);
   }
 
+  @Override
+  public Object getVariable(final String name) {
+    final Map<String, Object> variables = getVariablesAsMap();
+    if (!variables.containsKey(name)) {
+      throw new ClientException(String.format("The variable %s is not available", name));
+    }
```

Comment:
:thinking: I'm wondering if throwing an exception is a good approach here. Previously, a user could handle an unavailable variable by doing `Map#getOrDefault(...)`. Now, they would have to wrap their call in a try-catch block which would add several lines of code.

Would returning an `Optional.of(variable)` be nicer here?

Reply:
I think that request a variable that is not available is an error, so in some way we should provide to the user that feedback. If we use `Optional` we don't differ from a variable that is available but with `null` as value.

Reply:
:+1: That's a good point. Additionally, this is a new method, so we wouldn't break any existing user code/expectations. All good from my side.